AWSPinpoint/AWSPinpointNotificationManager.m

```diff
@@ -153,56 +152,60 @@ - (void)handleNotificationReceived:(UIApplication *) app
+            [self addGlobalEventSourceMetadataForNotification:userInfo];
+            [self recordMessageOpenedEventForNotification:userInfo
+                                           withIdentifier:nil];
+            if (shouldHandleNotificationDeepLink) {
+                [self handleNotificationDeepLinkForNotification:userInfo];
+            }
+            break;
+        }
+        case AWSPinpointPushActionTypeReceivedBackground: {
+            AWSDDLogVerbose(@"Received notification with app on background.");
+            [self addGlobalEventSourceMetadataForNotification:userInfo];
+            [self recordMessageReceivedEventForNotification:userInfo
+                                          withEventType:AWSEventTypeReceivedBackground];
+            break;
+        }
+        case AWSPinpointPushActionTypeReceivedForeground: {
+            AWSDDLogVerbose(@"Received notification with app on foreground.");
+
+            // Not adding global event source metadata because if the app session is already running,
+            // the session should not contribute to the new push notification that is being received
+            [self recordMessageReceivedEventForNotification:userInfo
+                                              withEventType:AWSEventTypeReceivedForeground];
+            break;
         }
-    } else if (state == UIApplicationStateBackground) {
-        AWSDDLogVerbose(@"Received notification with app on background.");
-        [self addGlobalEventSourceMetadataForNotification:userInfo];
-        [self recordMessageReceivedEventForNotification:userInfo
-                                              eventType:AWSEventTypeReceivedBackground
-                                   withApplicationState:state];
-    } else {
-        AWSDDLogVerbose(@"Received notification with app on foreground.");
-        [self recordMessageReceivedEventForNotification:userInfo
-                                              eventType:AWSEventTypeReceivedForeground
-                                   withApplicationState:state];
     }
 }
 
 #pragma mark - Event recorders
 - (void)recordMessageReceivedEventForNotification:(NSDictionary *) userInfo
-                                        eventType:(NSString*) eventType
-                             withApplicationState:(UIApplicationState) state {
+                                    withEventType:(NSString *) eventType {
     //Silent notification
     AWSPinpointEvent *pushNotificationEvent = [self.context.analyticsClient createEventWithEventType:eventType];
     
-    [self addApplicationStateAttributeToEvent:pushNotificationEvent
-                         withApplicationState:state];
     [self addEventSourceMetadataForEvent:pushNotificationEvent
                      withNotification:userInfo];
-    
+    [self addApplicationStateAttributeToEvent:pushNotificationEvent
+                         withApplicationState:[[UIApplication sharedApplication] applicationState]];
```

Comment:
It's pretty rare that the userInfo would have keys of UIApplicationStateActive, UIApplicationStateInactive, or UIApplicationStateBackground, but i prefer that we revert the ordering of these calls.

Reply:
Will move